lib/hijack.rb

```diff
@@ -53,10 +53,10 @@ def hijack(info: nil, &blk)
           # this trick avoids double render, also avoids any litter that the controller hooks
           # place on the response
           instance = controller_class.new
-          response = ActionDispatch::Response.new
-          instance.response = response
+          response = ActionDispatch::Response.new.tap { _1.request = request_copy }
+          instance.set_response!(response)
+          instance.set_request!(request_copy)
```

Comment:
Aren't `_1.request = request_copy` and `instance.set_request!(request_copy)` doing the same thing?

Reply:
That’s what I was thinking at first, and `instance.set_request!(request_copy)` wasn’t there initially. But some of the code was failing. I don’t remember exactly why, but IIRC, sometimes things are delegated to the response object, and sometimes it’s directly in the request object set on the controller.